docs/installation/server-config/security.rst

```diff
@@ -132,28 +133,130 @@ Configuring Authorization of KSQL with Kafka ACLs
+
+Principal
+    An authenticated user or group. For example, ``"user: Fred"`` or ``"group: fraud"``.
 
-This section uses the terminology used by the :ref:`Kafka Authorizer <kafka_authorization>` (``SimpleAclAuthorizer``)
-to describe the required ACLs. Each ACL is made up of these parts:
+Permission
+    Defines if the ACL allows (``ALLOW``) or denies (``DENY``) access to the resource.
+
+Operation
+    The operation that is performed on the resource, for example ``READ``.
 
 Resource
     A resource is comprised of a resource type and resource name:
 
     - ``RESOURCE_TYPE``, for example ``TOPIC`` or consumer ``GROUP``.
-    - Resource name, where the name is either specific, e.g. ``users``, or the wildcard ``*``, meaning all resources of this type.
+    - Resource name, e.g. the name of a topic or a consumer-group.
 
-    The ``CLUSTER`` resource type does not require a resource name because it refers to the entire Kafka cluster.
+ResourcePattern
+    A resource pattern matches zero or more Resources and is comprised of a resource type, a resource name and a pattern type.
 
-Operation
-    The operation that is performed on the resource, for example ``READ``.
+    - ``RESOURCE_TYPE``, for example ``TOPIC`` or consumer ``GROUP``. The pattern will only match resources of the same resource type.
+    - Resource name. How the pattern uses the name to match Resources is dependant on the pattern type.
+    - ``PATTERN_TYPE``, controls how the pattern matches a Resource's name to the patterns. Valid values are:
+
+        - ``LITERAL`` pattern types match the name of a resource exactly, or, in the case of the special wildcard resource name `*`, resources of any name.
+        - ``PREFIXED`` pattern types match when the resource's name is prefixed with the pattern's name.
+
+    The ``CLUSTER`` resource type is implicitly a literal pattern with a constant name because it refers to the entire Kafka cluster.
+
+An literal example might be ``ALLOW`` ``user Fred`` to ``READ`` the ``TOPIC`` with the ``LITERAL`` name ``users``.
+
+An prefixed example might be ``ALLOW`` ``user Fred`` to ``READ`` any ``TOPIC`` whose name is ``PREFIXED`` with ``fraud-``.
+
+The ACLs described below list a ``RESOURCE_TYPE``, resource name, ``PATTERN_TYPE``, and ``OPERATION``.
+All ACLs described are ``ALLOW`` ACLs, where the principal is the user the KSQL server has authenticated as,
+with the Apache Kafka cluster, or an appropriate group that includes the authenticated KSQL user.
+
+.. tip:: For more information about ACLs see :ref:`kafka_authorization` and for more information about interactive and
+non-interactive queries, see :ref:`restrict-ksql-interactive`.
+
+Required ACLs
+^^^^^^^^^^^^^
+
+The ACLs required are the same for both :ref:`Interactive and non-interactive (headless) KSQL clusters <restrict-ksql-interactive>`:
+
+KSQL require these ACLs:
```

Comment:
- require -> requires
- Please clarify that KSQL requires these ACLs *for what*?  For basic operation?  For ...?

Reply:
'to work' :D